packages/nimble-components/src/banner/template.ts

```diff
@@ -0,0 +1,42 @@
+import { html, slotted, when } from '@microsoft/fast-element';
+import { DesignSystem } from '@microsoft/fast-foundation';
+import type { Banner } from '.';
+import { Button } from '../button';
+import { IconExclamationMark } from '../icons/exclamation-mark';
+import { IconInfo } from '../icons/info';
+import { IconTriangleFilled } from '../icons/triangle-filled';
+import { IconXmark } from '../icons/xmark';
+
+// prettier-ignore
+export const template = html<Banner>`
+    <template role="status" aria-label=${x => x.getAriaLabel()}>
+        <div class="container">
+            <div class="icon">
+                ${when(x => x.severity === 'error', html`
+                    <${DesignSystem.tagFor(IconExclamationMark)}></${DesignSystem.tagFor(IconExclamationMark)}>
+                `)}
+                ${when(x => x.severity === 'warning', html`
+                    <${DesignSystem.tagFor(IconTriangleFilled)}></${DesignSystem.tagFor(IconTriangleFilled)}>
+                `)}
+                ${when(x => x.severity === 'info', html`
+                    <${DesignSystem.tagFor(IconInfo)}></${DesignSystem.tagFor(IconInfo)}>
+                `)}
+            </div>
+            <div class="text">
+                <slot name="title" ${slotted('slottedTitle')}></slot>
```

Comment:
Chatted offline but due to the slotted directive being implemented with the [slotchange event](https://developer.mozilla.org/en-US/docs/Web/API/HTMLSlotElement/slotchange_event) it does not trigger the update path for updating the aria-label.

Discussed going down the path of having the container div be the actual aria role and using aria-labelledby instead of serializing the slot content and copying it to the aria-label.

Reply:
Made changes as discussed.